File: firebase/functions/src/entity/Book.ts
import { Entity, PrimaryGeneratedColumn, Column, BaseEntity, OneToMany,} from "typeorm";
import { BookLibrary } from "./BookLibrary";

@Entity()
export class Book extends BaseEntity {
    @PrimaryGeneratedColumn()
    id: number;

    @Column({
        length: 13, 
        nullable: true
    }) isbn13: string;

    @Column({
        length: 10, 
        nullable: true
    }) isbn10: string; 

    @Column({
        nullable: true}) 
    dewey: string;

    @Column() 
    title: string;

    @Column() 
    author: string;

    @Column({
        default: 0, 
        nullable: true
    }) pages: number; 

    @Column({
        nullable: true
    }) 
    lang: string;

    @Column({
        nullable: true
    }) 
    image: string; 
   
    @OneToMany(type => BookLibrary, bookLibrary => bookLibrary.book)
    bookLibrary: BookLibrary[]; 


}

File: lib/model/book.dart
class Book {
  String image;
  int pages;
  String author;
  String isbn13;
  String isbn10;
  String dewey;
  int id;
  String title;
  String language;

  Book(this.image, this.pages, this.author, this.isbn13, this.isbn10,
      this.dewey, this.id, this.title, this.language);

  factory Book.fromJson(Map<dynamic, dynamic> record) {
    return Book(
        record['image'],
        record['pages'],
        record['author'],
        record['isbn13'],
        record['isbn10'],
        record['dewey'],
        record['id'],
        record['title'],
        record['lang']);
  }

  factory Book.fromApiJson(dynamic json) {
    // get book
    json = json['book'];

    // parse author
    var authors = json['authors'] != null ? List.from(json['authors']) : null;
    var author = authors.first;

    return Book(
        json['image'] as String,
        json['pages'] as int,
        author as String,
        json['isbn13'] as String,
        json['isbn'] as String,
        json['dewey_decimal'] as String,
        null,
        json['title'],
        json['language'] as String);
  }

  String get bookImg {
    return image != null ? image : '';
  }

  String get isbn_10 {
    return isbn10 != null ? isbn10 : '';
  }

  String get isbn_13 {
    return isbn13 != null ? isbn13 : '';
  }

  String get deweyd {
    return dewey != null ? dewey : '';
  }

  String get bookTitle {
    return title != null ? title : '';
  }

  String get bookAuthor {
    return author != null ? author : '';
  }

  int get pageCount {
    return pages != null ? pages : '';
  }

  String get bookLang {
    return language != null ? language : '';
  }
}
